src/main/java/io/cryostat/rules/RuleRegistry.java

```diff
@@ -179,6 +179,16 @@ public void deleteRules(ServiceRef serviceRef) throws IOException {
         }
     }
 
+    public void enableRule(Rule rule, boolean enabled) {
+        String name = rule.getName();
+        if (enabled && !rule.isEnabled()) {
+            getRule(name).get().setEnabled(enabled);
+            emit(RuleEvent.ADDED, rule);
```

Comment:
If I'm reading this correctly, the intent of this method is so that (re)enabling a rule, which can also happen when it's first created, emits an ADDED event. But disabling a rule does not emit an event. Enabling an already-enabled rule does not emit an event.

Would it make sense for this to be "symmetric" and also emit a REMOVED event when an enabled rule becomes disabled?

Or perhaps there can simply be an UPDATED/MODIFIED event too? Since the event contains the Rule itself there is no need for the event type to reflect the enabled state, that can just be read from the Rule itself.

Reply:
Created the UPDATED event where the rules are activated or deactivated when enabled/disabled.